main.go

```diff
@@ -2,16 +2,115 @@ package main
 
 import (
 	"fmt"
+	"os"
+	"runtime"
+	"strings"
 	"time"
+
+	"github.com/urfave/cli/v2"
 )
 
 var (
 	// these variables are populated by Goreleaser when releasing
 	version = "unknown"
 	commit  = "-dirty-"
 	date    = time.Now().Format("2006-01-02")
+
+	// TODO: Adjust app name
+	appName     = "go-bootstrap"
+	appLongName = "a generic bootstrapping project"
+
+	// TODO: Adjust or clear env var prefix
+	// envPrefix is the global prefix to use for the keys in environment variables
+	envPrefix = "BOOTSTRAP"
 )
 
 func main() {
-	fmt.Printf("version %s, commit %s, date %s\n", version, commit, date)
+	app := newApp()
+	_ = app.Run(os.Args)
```

Comment:
I have one other nitpicky/nice-to-have feature request. We could add signal handling by default with `signal.NotifyContext`

Reply:
I'm not sure what you mean. By default, signal handling already works? (e.g. CTRL-C, terminate)
Or did you mean to execute `After` in a CLI (sub)command?

Reply:
I took me a while to workaround some limitations with CLI lib but I think I got it working

Reply:
Sorry I'm a bit late with my reply.

That got a lot more complicated then I thought. What I had in mind was to just add `signal.NotifyContext` to the context given to `RunContext`. The command should then be able to handle context cancellation. This gives us a simple way to gracefully shut down.

Reply:
I thought with the current implementation I am giving the commands an easy way to gracefully shutdown.